### PR TITLE
STAR add suggested parameter for Arriba --alignSJstitchMismatchNmax and add softclip option --chimOutType 

### PR DESCRIPTION
**Pull request by JasperO98**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

I have added the parameters `--alignSJstitchMismatchNmax` and `--chimOutType` parameters are missing, which are recommended parameters for the fusion gene detection tool Arriba. All recommended parameters for Arriba are available, see [documentation](https://arriba.readthedocs.io/en/latest/workflow/).  This pull request is related to #4848 where Arriba can control STAR to automatically use the correct parameters. This had some issues so it would be better to expose the arguments in STAR, which are added here.

### CODE REVIEW

File: tools/rgrnastar/rg_rnaStar.xml [hunk 1]
```diff
@@ -348,7 +362,8 @@
         help="Choose if and how chimeric alignments should be reported. STAR-Fusion users should select the 'Junctions' option and use the resulting tabular dataset as input to STAR-Fusion. Everyone else: note that selecting 'WithinBAM' or 'WithinBAM Junctions' disables the --chimMultimapNmax setting in the algorithmic parameters section below (the tool will only consider uniquely mapped reads in the search for chimeric alignments). If you disable the reporting of chimeric alignments here, then all chimeric alignment settings in the algorithmic parameters section below will be ignored.">
             <option value="">Don't report chimeric alignments</option>
             <option value="Junctions">As separate tabular "Junctions" output (Junctions)</option>
-            <option value="WithinBAM">Within the BAM output (together with regular alignments; WithinBAM)</option>
+            <option value="WithinBAM HardClip">Within the BAM output (together with regular alignments; WithinBAM HardClip) hard-clipping in the CIGAR for supplemental chimeric alignments</option>
```
Comment (bgruening) [inline tools/rgrnastar/rg_rnaStar.xml:365]: Does the previous "WithinBam" not exist anymore?

**Review by bgruening — COMMENTED** [review]
Thanks @JasperO98 two small comments online.

File: tools/rgrnastar/rg_rnaStar.xml (hunk 1 above)
Comment (JasperO98) [reply tools/rgrnastar/rg_rnaStar.xml:365]: Oops it does, but `WithinBam HardClip` is now the default in STAR.

Comment (bgruening) [reply tools/rgrnastar/rg_rnaStar.xml:366]: You can set the default in Galaxy as well, with "selected" I belief.

Comment (JasperO98) [reply tools/rgrnastar/rg_rnaStar.xml:366]: The option `WithinBAM Junctions` is depecrated I think. At least it is not described in the manual of 2.7.0a or 2.7.10b.

- 2.7.0a: https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf
- 2.7.10b: https://raw.githubusercontent.com/alexdobin/STAR/master/doc/STARmanual.pdf

Should it be removed?

File: tools/rgrnastar/rg_rnaStar.xml [hunk 2]
```diff
@@ -145,6 +145,19 @@
             --chimMultimapScoreRange 10
             --chimMultimapNmax 10
             --chimNonchimScoreDropMin 10
+        
+        #elif str( $algo.params.settingsType ) == 'arriba':
+            ## Preset parameters for Arriba
```
[the remark this answers is not in the document]
Comment (JasperO98) [reply tools/rgrnastar/rg_rnaStar.xml:150]: Fixed in [5bba325](https://github.com/galaxyproject/tools-iuc/pull/4889/commits/5bba325d7ced8a750b20f05656dd138e799facb4)

File: tools/rgrnastar/rg_rnaStar.xml (hunk 1 above)
Comment (bgruening) [reply tools/rgrnastar/rg_rnaStar.xml:366]: Lets add a deprecated text to it and remove it in the next version maybe? It has a test as far as I can see and it still seems to work. But if its not mentioned it should not be the default I assume :)

Comment (JasperO98) [reply tools/rgrnastar/rg_rnaStar.xml:366]: Fixed in https://github.com/galaxyproject/tools-iuc/pull/4889/commits/3503bfe4e531e304979699462e17f24830b8da1f

Comment (bgruening) [reply tools/rgrnastar/rg_rnaStar.xml:366]: Do you want to make hardclip the default?

Comment (JasperO98) [reply tools/rgrnastar/rg_rnaStar.xml:366]: No it is fine this way if you are not doing fusion gene detection you don't have a need for this option.

**Review by bgruening — APPROVED** [review]

**Comment by JasperO98** [comment]
Can we merge this PR now?

**Comment by bgruening** [comment]
Thanks a lot @JasperO98!

**Pull request merged**